### PR TITLE
Fix Issue #743, Updated word2vec.n_similarities and test_word2vec.testSimilarities methods

### PR DESCRIPTION
**Pull request by pranay360**
Changes requested in PR #882 completed
1. word2vec.n_similarities now raise a ZeroDivisionError when atleast 1 empty list is passed to it.
2. In test_word2vec.testSimilarities added testcases involving empty lists to check if ZeroDivisionError is raised.
3. pep8 followed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,10 @@
 Changes
 =======
 
-0.13.2, 2016-08-19
-* export_phrases in Phrases model changed. Fixed issue #794 and added test cases in test/test_phrases.py(@AadityaJ,
-[#879](https://github.com/RaRe-Technologies/gensim/pull/879))
```
Comment (tmylk) [inline CHANGELOG.md]: Please don't delete from changelog of phrases fix.

File: gensim/models/word2vec.py [hunk 2]
```diff
@@ -1539,9 +1539,15 @@ def n_similarity(self, ws1, ws2):
...
-        v2 = [self[word] for word in ws2]
-        return dot(matutils.unitvec(array(v1).mean(axis=0)), matutils.unitvec(array(v2).mean(axis=0)))
+        if len(ws1) > 0 and len(ws2) > 0:
+            v1 = [self[word] for word in ws1]
+            v2 = [self[word] for word in ws2]
+            return dot(matutils.unitvec(array(v1).mean(axis=0)),
+                       matutils.unitvec(array(v2).mean(axis=0)))
+        else:
+            raise ZeroDivisionError('Atleast one of the passed list is empty.')
+            return
```
Comment (tmylk) [inline gensim/models/word2vec.py]: no need to return.

It is more concise to just have one statement above
```
if not (len(ws1) > 0 and len(ws2) > 0) :
    raise ZeroDivisionError('Atleast one of the passed list is empty.')
```

File: CHANGELOG.md (hunk 1 above)
Comment (pranay360) [inline CHANGELOG.md]: I didn't delete it, I had a previous build in which that entry was not in the changelog, anyway I added it. Thank You for marking out the changes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,10 +2,10 @@ Changes
 =======
 
 0.13.2, 2016-08-19
-* export_phrases in Phrases model changed. Fixed issue #794 and added test cases in test/test_phrases.py(@AadityaJ,
-[#879](https://github.com/RaRe-Technologies/gensim/pull/879))
-    - bigram construction can now support multiple bigrams within one sentence
-* wordtopics has changed to word_topics in ldamallet, and fixed issue #764. (@bhargavvader, [#771](https://github.com/RaRe-Technologies/gensim/pull/771))
+
+* Fixed issue #743 , In word2vec's n_similarity method if atleast one empty list is passed ZeroDivisionError is raised.
+* export_phrases in Phrases model changed. Fixed issue #794 and added test cases in test/test_phrases.py(@AadityaJ, #879) bigram construction can now support multiple bigrams within one sentence
```
Comment (tmylk) [inline CHANGELOG.md]: Please add to future release 0.13.3

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
+
+* Fixed issue #743 , In word2vec's n_similarity method if atleast one empty list is passed ZeroDivisionError is raised, added test cases in test/test_word2vec.py(@pranay360, #883)
+
 0.13.2, 2016-08-19
-* export_phrases in Phrases model changed. Fixed issue #794 and added test cases in test/test_phrases.py(@AadityaJ,
-[#879](https://github.com/RaRe-Technologies/gensim/pull/879))
-    - bigram construction can now support multiple bigrams within one sentence
-* wordtopics has changed to word_topics in ldamallet, and fixed issue #764. (@bhargavvader, [#771](https://github.com/RaRe-Technologies/gensim/pull/771))
+
+* export_phrases in Phrases model changed. Fixed issue #794 and added test cases in test/test_phrases.py(@AadityaJ, #879) bigram construction can now support multiple bigrams within one sentence
```
Comment (tmylk) [inline CHANGELOG.md]: Is this change necessary?

Comment (pranay360) [inline CHANGELOG.md]: are you talking about fixing the issue?

Comment (tmylk) [inline CHANGELOG.md]: This pr is about #743. There is no need to change other parts of changelog.md

File: gensim/models/word2vec.py [hunk 5]
```diff
@@ -1539,9 +1539,13 @@ def n_similarity(self, ws1, ws2):
           True
 
         """
+        if not(len(ws1) and len(ws2)):
+            raise ZeroDivisionError('Atleast one of the passed list is empty.')
         v1 = [self[word] for word in ws1]
         v2 = [self[word] for word in ws2]
-        return dot(matutils.unitvec(array(v1).mean(axis=0)), matutils.unitvec(array(v2).mean(axis=0)))
+        return dot(matutils.unitvec(array(v1).mean(axis=0)),
```
Comment (piskvorky) [inline gensim/models/word2vec.py]: We don't use vertical indent in gensim -- change to normal hanging indent (see PEP8: all statements on new lines, one level of indent).

File: gensim/models/word2vec.py [hunk 6]
```diff
@@ -1539,9 +1539,13 @@ def n_similarity(self, ws1, ws2):
           True
 
         """
+        if not(len(ws1) and len(ws2)):
+            raise ZeroDivisionError('Atleast one of the passed list is empty.')
```
Comment (piskvorky) [inline gensim/models/word2vec.py]: If input is a list, idiomatic Python is: `if not ws1 or not ws2:`.

**Comment by piskvorky** [comment]
Code style changes requested; please fix.

This has already been merged, so probably in another PR. Cheers.

File: gensim/models/word2vec.py (hunk 2 above)
Comment (piskvorky) [inline gensim/models/word2vec.py]: Agreed with @gojomo .
